src/platform/backends/hyperv_api/virtdisk/virtdisk_snapshot.cpp

```diff
@@ -0,0 +1,284 @@
+/*
+ * Copyright (C) Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include <hyperv_api/virtdisk/virtdisk_snapshot.h>
+
+#include <hyperv_api/virtdisk/virtdisk_wrapper.h>
+
+#include <multipass/exceptions/formatted_exception_base.h>
+#include <multipass/file_ops.h>
+#include <multipass/virtual_machine.h>
+#include <multipass/virtual_machine_description.h>
+
+namespace
+{
+constexpr auto log_category = "virtdisk-snapshot";
+}
+
+namespace multipass::hyperv::virtdisk
+{
+
+struct CreateVirtdiskSnapshotError : public FormattedExceptionBase<std::system_error>
+{
+    using FormattedExceptionBase<std::system_error>::FormattedExceptionBase;
```

Comment:
You could remove `public` and the template type (both are implied):

```suggestion
struct CreateVirtdiskSnapshotError : FormattedExceptionBase<std::system_error>
{
    using FormattedExceptionBase::FormattedExceptionBase;
```

Reply:
Done